lib/cosmos/tools/cmd_tlm_server/gui/interfaces_tab.rb

```diff
@@ -154,7 +154,7 @@ def create_button(name, interface, interface_name)
       button = Qt::PushButton.new(button_text)
       if name == ROUTERS
         button.connect(SIGNAL('clicked()')) do
-          if interface.thread
+          if CmdTlmServer.routers.all[interface_name].thread
```

Comment:
it could probably be cleaner but its ok to use the actual interface, when creating the button.   The button handlers need to use the current interface not the old one though.